File: retroauth-core/src/main/java/com/andretietz/retroauth/TokenTypeFactory.java
package com.andretietz.retroauth;

public interface TokenTypeFactory<TOKEN_TYPE> {
    /**
     * Creates a token type object. This doe
     *
     * @param annotationValues The values from the {@link Authenticated} annotation
     * @return a token type.
     */
    TOKEN_TYPE create(int[] annotationValues);
}

File: retroauth/src/main/java/com/andretietz/retroauth/TokenTypeFactory.kt
package com.andretietz.retroauth

interface TokenTypeFactory<TOKEN_TYPE : Any> {
    /**
     * Creates a token type object. This doe
     *
     * @param annotationValues The values from the [Authenticated] annotation
     * @return a token type.
     */
    fun create(annotationValues: IntArray): TOKEN_TYPE
}
